apps/api/src/domain/models/digital-text/entities/digital-text.add-content-to-page.spec.ts

```diff
@@ -0,0 +1,145 @@
+                idOfPageThatDoesntExist,
+                textContent,
+                languageCode
+            );
+
+            assertErrorAsExpected(
+                result,
+                new FailedToUpdateDigitalTextPageError(
+                    idOfPageThatDoesntExist,
+                    existingDigitalText.id,
+                    [
+                        new CannotAddContentToMissingPageError(
+                            idOfPageThatDoesntExist,
+                            existingDigitalText.id
+                        ),
+                    ]
+                )
+            );
+        });
+    });
+
+    describe(`when the page already has content`, () => {
+        it(`should return the approprate error`, () => {
+            const digitalTextWithContentForTargetPage = existingDigitalText.addContentToPage(
+                existingPageIdentifier,
+                textContent,
+                languageCode
+            ) as DigitalText;
+
+            const result = digitalTextWithContentForTargetPage.addContentToPage(
+                existingPageIdentifier,
+                'New content',
+                languageCode
+            );
+
+            const existingPage = digitalTextWithContentForTargetPage.getPage(
+                existingPageIdentifier
+            ) as DigitalTextPage; // and not `NotFound`
+
+            assertErrorAsExpected(
+                result,
+                new FailedToUpdateDigitalTextPageError(
+                    existingPageIdentifier,
+                    existingDigitalText.id,
+                    [
+                        new CannotOverwritePageContentError(
+                            existingPageIdentifier,
+                            existingPage.getContent() as MultilingualText // and not `NotFound`
+                        ),
+                    ]
+                )
+            );
+        });
+    });
+
+    describe(`when the input is ill formed`, () => {
+        describe(`when the page identifier is an empty string`, () => {
+            it(`should return the appropriate error`, () => {
+                const result = existingDigitalText.addContentToPage('', textContent, languageCode);
+
+                // This is actually a "cannot add content for missing page"
+                expect(result).toBeInstanceOf(InternalError);
+            });
+        });
+
+        describe(`when the text content for the page is an empty string`, () => {
+            it(`should return the appropriate error`, () => {
+                const result = existingDigitalText.addContentToPage(
+                    existingPageIdentifier,
+                    '',
+                    languageCode
+                );
+
+                assertErrorAsExpected(
+                    result,
+                    new InvariantValidationError(existingDigitalText.getCompositeIdentifier(), [])
+                );
+            });
+        });
+
+        // TODO Think about whether we really need this
```

Comment:
We are testing the method, not the command FSA that comes over the wire. We know that we must adapt the command FSA to a method call in the command handler and that there we will have full type-safety. The only reason to test this directly is if we do not trust TypeScript. I'd say let's skip this and if we are ever bitten by it, we can come back and add regression tests.